File: odd_even_pairs.java
import java.io.*;
import java.util.*;
class HE
{
	public static void main(String[] args)
	{
	    Scanner sc=new Scanner(System.in);
	    int n,i,j,odd=0,even=0;
	    n=sc.nextInt();
	    int arr[]=new int[n];
	    for(i=0;i<n;i++)
	    {
	        arr[i]=sc.nextInt();
	    }
	    for(i=0;i<n;i++)
	    {
	        for(j=i+1;j<n;j++)
	        {
	            if((arr[i]+arr[j])%2==0)
	            {
	                even++;
	            }
	            else
	            {
	                odd++;
	            }
	        }
	    }
	    System.out.println(Math.abs(odd-even));
	}
}

File: Odd_Even_Pairs.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
	int N;
	scanf("%d", &N);
	int arr[N];
	for(int i = 0; i < N; i++)
	{
		scanf("%d", &arr[i]);
	}
	int odd_sum = 0;
	int even_sum = 0;
	int odd = 0, even = 0;
	for(int i = 0; i < N; i++)
	{
		if(arr[i] % 2 == 0)
		{
			odd_sum += odd;
			even_sum += even;
			even++;
		}
		else
		{
			odd_sum += even;
			even_sum += odd;
			odd++;
		}
	}
	printf("%d\n", abs(odd_sum - even_sum));
	return 0;
}